services/dataService.js

```diff
@@ -0,0 +1,67 @@
+const productMap = [];
```

Comment:
Don't initialize this as an array, should be an object or `null`.

Reply:
Can't remember why in the world I intialized it as an array. Thanks! :)